webpack.config.js

```diff
@@ -22,151 +24,114 @@ const SCOPE_VERSION = JSON.stringify(versionHash)
+	}, {}),
+	...glob.sync('src/mixins/*/index.js').reduce((acc, item) => {
+		const name = item
+			.replace('/index.js', '')
+			.replace('src/mixins/', 'Mixins/')
+		acc[name] = path.join(__dirname, item)
+		return acc
+	}, {}),
+}
+
+webpackConfig.output = {
+	path: path.resolve(__dirname, './dist'),
+	publicPath: '/dist/',
+	filename: `[name].${libraryTarget}.js`,
+	libraryTarget: 'umd',
+	library: {
+		type: libraryTarget,
+		name: ['NextcloudVue', '[name]'],
 	},
-	externals: [nodeExternals()],
-	module: {
-		rules: [
-			{
-				test: /\.css$/,
-				use: ['style-loader', 'css-loader'],
-			},
-			{
-				test: /\.scss$/,
-				use: [
-					'style-loader',
-					'css-loader',
-					'resolve-url-loader',
-					{
-						loader: 'sass-loader',
-						options: {
-							additionalData: `@use 'sass:math'; $scope_version:${SCOPE_VERSION}; @import 'variables'; @import 'material-icons';`,
-							/**
-							 * ! needed for resolve-url-loader
-							 */
-							sourceMap: true,
-							sassOptions: {
-								sourceMapContents: false,
-								includePaths: [
-									path.resolve(__dirname, './src/assets'),
-								],
-							},
-						},
-					},
-				],
-			},
-			{
-				test: /\.vue$/,
-				loader: 'vue-loader',
-			},
-			{
-				test: /\.js$/,
-				loader: 'babel-loader',
-				exclude: BabelLoaderExcludeNodeModulesExcept([
-					'@nextcloud/calendar-js',
-					'tributejs',
-				]),
-			},
-			{
-				test: /\.(png|jpg|gif|eot|ttf|woff|woff2)$/i,
-				loader: 'url-loader',
-			},
-			{
-				test: /\.svg/,
-				type: 'asset/inline',
-			},
-			{
-				resourceQuery: /raw/,
-				type: 'asset/source',
+	umdNamedDefine: true,
+}
+
+webpackConfig.externals = [nodeExternals()]
+
+webpackRules.RULE_SCSS = {
+	test: /\.scss$/,
+	use: [
+		'style-loader',
+		'css-loader',
+		'resolve-url-loader',
+		{
+			loader: 'sass-loader',
+			options: {
+				additionalData: `@use 'sass:math'; $scope_version:${SCOPE_VERSION}; @import 'variables'; @import 'material-icons';`,
+				/**
+				 * ! needed for resolve-url-loader
+				 */
+				sourceMap: true,
+				sassOptions: {
+					sourceMapContents: false,
+					includePaths: [
+						path.resolve(__dirname, './src/assets'),
+					],
+				},
 			},
-		],
-	},
-	plugins: [
-		new VueLoaderPlugin(),
-	],
-	resolve: {
-		extensions: ['*', '.js', '.vue'],
-		symlinks: false,
-		// Alias the default server assets path to link to local one
-		alias: {
-			'/core/css/../img': path.join(__dirname, 'styleguide/assets/img'),
-			'/core/css/../fonts': path.join(__dirname, 'styleguide/assets/fonts'),
 		},
-	},
+	],
+}
+
+webpackRules.RULE_JS.exclude = BabelLoaderExcludeNodeModulesExcept([
+	'@nextcloud/calendar-js',
+	'tributejs',
+])
+
+webpackConfig.module.rules = Object.values(webpackRules)
+
+// Alias the default server assets path to link to local one
+webpackConfig.resolve.alias = {
+	'/core/css/../img': path.join(__dirname, 'styleguide/assets/img'),
+	'/core/css/../fonts': path.join(__dirname, 'styleguide/assets/fonts'),
```

Comment:
is this still needed?

Reply:
Seems to work fine without it. I removed it.